test/VaultRegistry.test.ts

```diff
@@ -1,445 +1,809 @@
+                                .registerVault(
+                                    this.anotherERC20Vault.address,
+                                    await this.ownerSigner.getAddress()
+                                );
+                        }
+                    );
+                    this.nftAnotherERC20 = Number(
+                        await this.subject.vaultsCount()
+                    );
+                    let newVaultsCount = Number(
+                        await this.subject.vaultsCount()
+                    );
+                    expect(newVaultsCount - oldVaultsCount).to.be.equal(1);
+                });
+            });
+        });
+    });
+
+    describe("#registerVault", () => {
+        it("binds minted ERC721 NFT to Vault address and transfers minted NFT to owner specified in args", async () => {
+            let newOwner = randomAddress();
+            expect(await this.subject.balanceOf(newOwner)).to.be.equal(0);
+            await withSigner(this.erc20VaultGovernance.address, async (s) => {
+                await this.subject
+                    .connect(s)
+                    .registerVault(this.anotherERC20Vault.address, newOwner);
+            });
+            expect(await this.subject.balanceOf(newOwner)).to.be.equal(1);
+        });
+        it("emits VaultRegistered event", async () => {
+            await withSigner(this.erc20VaultGovernance.address, async (s) => {
+                await expect(
+                    this.subject
+                        .connect(s)
+                        .registerVault(
+                            this.anotherERC20Vault.address,
+                            this.ownerSigner.address
+                        )
+                ).to.emit(this.subject, "VaultRegistered");
+            });
+        });
+
+        describe("properties", () => {
+            it("@property: minted NFT equals to vaultRegistry#vaultsCount", async () => {
+                let newOwner = randomAddress();
+                expect(await this.subject.balanceOf(newOwner)).to.be.equal(0);
+                await withSigner(
+                    this.erc20VaultGovernance.address,
+                    async (s) => {
+                        await this.subject
+                            .connect(s)
+                            .registerVault(
+                                this.anotherERC20Vault.address,
+                                newOwner
+                            );
+                    }
+                );
+                expect(await this.subject.balanceOf(newOwner)).to.be.equal(1);
+                expect(
+                    await this.subject.ownerOf(
+                        (await this.subject.vaultsCount()).toNumber()
+                    )
+                ).to.be.equal(newOwner);
+            });
+        });
+
+        describe("access control:", () => {
+            it("allowed: any VaultGovernance registered in ProtocolGovernance", async () => {
```

Comment:
I'd rename it:
allowed: any address with CREATE_VAULT permission (btw update to the latest `main` I think here VAULT_GOVERNANCE permission is used which is deprecated)

And the test it accordingly